shard.go

```diff
@@ -1,33 +1,48 @@
 package influxdb
 
 import (
-	"encoding/json"
+	"encoding/binary"
 	"errors"
 	"fmt"
+	"math"
+	"sort"
 	"time"
-	"unsafe"
 
 	"github.com/boltdb/bolt"
 )
 
-// Shard represents the physical storage for a given time range.
-type Shard struct {
+// ShardGroup represents a group of shards created for a single time range.
+type ShardGroup struct {
 	ID        uint64    `json:"id,omitempty"`
 	StartTime time.Time `json:"startTime,omitempty"`
 	EndTime   time.Time `json:"endTime,omitempty"`
+	Shards    []*Shard  `json:"shards,omitempty"`
+}
 
-	replicaN    []uint64 // replication factor
-	dataNodeIDs []uint64 // owner nodes
+// close closes all shards.
+func (g *ShardGroup) close() {
+	for _, sh := range g.Shards {
+		_ = sh.close()
+	}
+}
+
+// Shard represents the physical storage for a given time range.
```

Comment:
I changed the comment. The shard data should probably be attached directly to the `Server` and then the `Shard` is simply a logical construct. I need to do more to rework the write series in later PRs. I can make those adjustments there.